### PR TITLE
Fix "all" of 1.0 issues.

### PR DESCRIPTION
**Pull request by Datseris**
final deprecation fix for 1., maybe!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,12 +24,8 @@ matrix:
...
     - test "x$TRAVIS_OS_NAME" = xosx -o "x$PYTHON" = xConda || (sudo apt-get -qq update && sudo apt-get install -y $PYTHON)
-script:
-    - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-    - julia -e 'Pkg.clone(pwd()); Pkg.build("PyCall")'
-    - julia -e 'import PyCall' # precompile
-    - julia -e 'Pkg.test("PyCall"; coverage=true)'
+
 after_success:
-    - julia -e 'cd(Pkg.dir("PyCall")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
+    - julia -e 'using Pkg; cd(joinpath(dirname(pathof(PyCall)), "..")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
```
Comment (fredrikekre) [inline .travis.yml]: `using PyCall` is needed if you want the `pathof` it. But this `cd` can be removed completely.

**Comment by Datseris** [comment]
I need some help from someone that develops PyCall. I have encountered a "real" error on 1.0:

https://travis-ci.org/JuliaPy/PyCall.jl/jobs/416296896#L727

```
  Test threw exception
  Expression: BigInt(o) == i
  MethodError: no method matching BigInt(::PyObject)
  Closest candidates are:
    BigInt() at gmp.jl:56
    BigInt(!Matched::BigInt) at gmp.jl:235
    BigInt(!Matched::Union{Int32, Int64}) at gmp.jl:266
```
If someone knows why, please push a fix :)

**Comment by stevengj** [comment]
@Datseris, should be fixed by 3e5831e45421cdedf16397e3653c308f7b4d68c0 … calls to `T(::PyObject)` should have been replaced by explicit `convert` calls in the 0.7 upgrade, but it looks like I missed that one.

**Comment by Datseris** [comment]
Thanks @stevengj . I am kind of a git noob, how do I incorporate this last commit into this branch, so that the tests pass and we see the cool green color?

**Comment by stevengj** [comment]
Would also be good to use the appveyor script from https://github.com/JuliaCI/Appveyor.jl (but add `julia_version: 0.6`)

**Comment by stevengj** [comment]
Assuming you have your `patch-1` branch locally:
```sh
git checkout master
git pull origin master
git checkout patch-1
git rebase master
git push https://github.com/Datseris/PyCall.jl +patch-1
```

**Comment by stevengj** [comment]
Or, since it is a 1-line change, just make the same change yourself in this PR.   (As long as your change is identical, there will be no merge conflict.)

**Comment by tkf** [comment]
I think you can copy the default script of Travis from here:

https://github.com/travis-ci/travis-build/blob/2b32563bba377708822352fbc3660a317ee38d28/lib/travis/build/script/julia.rb#L81

which is what I suggested for IJulia.jl:

https://github.com/JuliaLang/IJulia.jl/pull/710/commits/5e1f1335793888bfb3a88033708507a8ff8e8911#diff-180360612c6b8c4ed830919bbb4dd459

Though I'm not sure this works.  Still waiting for AppVeyor to finish the build :)

**Comment by stevengj** [comment]
The AppVeyor script is broken, but it is passing on Travis, so I will go ahead and merge.

**Comment by stevengj** [comment]
See #524 for a new appveyor script.

**Pull request merged**